### ISSUE TITLE
NVDA does not read line numbers in Visual Studio

### ISSUE THREAD

**Issue by Shyopinggo** [nvaccess/nvda#14141]
Hi!
Earlier NVDA read line numbers in Visual Studio even if read line numbers was disabled in settings. I don’t know was it a bug or feature, but it was very convenient, because it was reading just a number and a text without saying “line 154 blablabla, line 1254 blablabla”.
After updating to the newest version NVDA does not read line numbers in Visual Studio anymore.
### Steps to reproduce:
(Read line numbers in settings is disabled)
1. Open any file in Visual Studio containing for example C# code.
2. Read a file pressing up and down arrows.
### Actual behavior:
NVDA reads the file without saying numbers before text.
### Expected behavior:
Before text NVDA says a number which corresponds a number of this line in the file.
#### NVDA installed/portable/running from source:
installed
#### NVDA version:
release 2022.2.2
#### Windows version:
Windows 10 Pro. Version: 10.0.19043, build: 19043
#### Name and version of other software in use when reproducing the issue:
Visual Studio Community 2022 version 17.3.3
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
It worked on erlyer versions.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by seanbudd** [comment]
I am surprised that 2022.2.2 no longer works, considering this feature was introduced in 2022.2.
It's possible something has changed in Visual Studio. Can you confirm 2022.2 still reports line numbers, but 2022.2.2 doesn't?

**Comment by Shyopinggo** [comment]
Hmm… Interesting. First, sorry, I didn’t mention that I’ve updated from 2022.1. I’ve just read “wat’s new” for 2022.2 and saw that there was really added this feature. But it worked fine for me in previous versions. So, here are results of my tests.
Tested on Visual Studio Community and Professional 2019 and 2022.
NVDA 2022.2 and 2022.2.2 (read line numbers enabled) says for example:
Line 323 var email = new EmailProcessor();
Line 324 email.Send();
NVDA 2022.2 and 2022.2.2 (read line numbers disabled) says for example:
var email = new EmailProcessor();
email.Send();
For example, NVDA 2021.3.4 (read line numbers no difference – same result) says:
323 var email = new EmailProcessor();
324 email.Send();

**Comment by seanbudd** [comment]
Closing this as it appears to be resolved as the setting is working as expected

**Comment by Shyopinggo** [comment]
Ok, I agree it can be closed. But is there way to make it work as previously? Now I either don’t hear numbers at all if the setting is disabled, or hear the word “line” before each number. Previously I heard only numbers. I have to read tones of source code in my work, and hearing word “line” every time dramatically slows down the process.

**Comment by seanbudd** [comment]
@Shyopinggo - This is standard behaviour for reporting line numbers in NVDA. I would suggest creating a new issue or GitHub discussion to propose changing the default behaviour for all line number reporting to something shorter.

**Comment by Shyopinggo** [comment]
Thanks a lot!

**Comment by CyrilleB79** [comment]
Side note:
In Jaws there is a feature allowing to customize the screenreader's messages: for example you can replace reporting "heading    level 3" by what you want, e.g. "heading 3". Maybe there is already a ticket for implementing such a feature in NVDA but I did not find it. If implemented, this feature could allow you to replace a message like "Line {number}" by "{number}".

So to summarize, you may want to open a ticket for this broader feature or comment there if it already exists.

**Comment by cary-rowen** [comment]
An easy way, you might be able to modify nvda.po's interface translation file to do this, but it's not recommended.

**Comment by DalisonCPU** [comment]
It seems that the problem with the configuration has happened again.
